subprojects/files/src/main/java/org/gradle/internal/file/impl/DefaultDeleter.java

```diff
@@ -63,9 +63,17 @@ public boolean deleteRecursively(File root, boolean followSymlinks) throws IOExc
     }
 
     @Override
-    public boolean cleanRecursively(File root, boolean followSymlinks) throws IOException {
-        if (root.exists() && shouldRemoveContentsOf(root, followSymlinks)) {
+    public boolean ensureEmptyDirectory(File root, boolean followSymlinks) throws IOException {
+        if (root.isDirectory()) {
```

Comment:
Don't we need to check if the root is a symlink and remove it? A symlink to a directory still returns `true` for `isDirectory()`.

Reply:
We don't if we adhere to the Javadoc, which says that we keep root symlinks pointing to directories.

Reply:
Ah, I got confused by the two Javadocs talking about symlinks. For the directory we have
```
a symlink pointing to an existing directory, then the linked directory's contents are removed recursively,
```

IIUC, currently the linked directory's contents are only removed if `followSymlinks` is `true`, right? Is this intended?

Reply:
Good catch. Fixed it and added a test.